wallet-new/src/Cardano/Wallet/Kernel/Actions.hs

```diff
@@ -210,20 +213,13 @@ withWalletWorker wai k = do
       pushWA = writeTMQueue tmq >=> \case
          True -> pure ()
          False -> Ex.throwM WalletWorkerExpiredError
-  -- Run the worker in the background, ensuring that any exceptions from it
-  -- get thrown to the current thread.
-  Ex.bracket
-     (liftIO $ do
-        as1 <- Async.async (walletWorker wai (STM.atomically getWA))
-        Async.link as1
-        pure as1)
-     (\as1 -> liftIO $ do
-        -- Prevent new input.
-        STM.atomically (closeTMQueue tmq)
-        -- Wait for the worker to finish, re-throwing any exceptions from it.
-        Async.wait as1)
-     (\_ -> k pushWA)
-
+  fmap snd $ withUnliftIO $ \(unlift :: UnliftIO m) -> Async.concurrently
+    -- Queue reader. If it dies, the writer dies too.
+    (walletWorker wai (STM.atomically getWA))
+    -- Queue writer. If it finishes, it closes the queue, causing the reader
+    -- to terminate normally. If it dies, forget about closing the queue, just
+    -- kill the reader.
```

Comment:
`Async.concurrently` takes care of this for us. No need to explicitly `Async.link` the reader thread.

I want to emphasise that the meaning has changed here. If this `k` action dies exceptionally, then whatever is in the queue will not be read and `interp`'d.

Reply:
(and I think that's a good thing)